### DISCUSSION TITLE
Setting up a Multipage CMS with Sulu

### DISCUSSION THREAD

**Discussion by brachycera** [sulu/sulu#6647] [Q&A] [answered]
I have a few Questions and I'm not sure if Sulu is the right tool for my requirements. Much appreciated if someone can point me in the right directions.

Is it possible to setup the Sulu Administration at e.g. `sulu.domainA.com`.

And is it then possible, to create content from the above Sulu Administration for multiple domains (via `webspaces`) e.g. `domainB.com`, `domainC.com` etc.

Thanks for any help!

**Comment by alexander-schranz** [comment, marked as answer]
> Is it possible to setup the Sulu Administration at e.g. sulu.domainA.com.

You can change the admin url this way: https://github.com/sulu/sulu-demo/pull/60/files. In the example it done by changing the path but you can also configure an own `host` and so make it only available only for a specific domain.

> And is it then possible, to create content from the above Sulu Administration for multiple domains (via webspaces) e.g. domainB.com, domainC.com etc.

As sulu is very flexible domains can be used in different ways. There are 3 scenarios (as webspace, as language or as own installation):

**Scenario A:**

Each "website" has its own domain but has shared content (snippets, contacts, media, ...). Then the right way is that you just create an own webspace for every website:

https://docs.sulu.io/en/2.4/cookbook/create-new-webspace.html

You configure each webspace with each domain in the `<url>` part of the webspace. Through the webspace switcher you can then change to the other webspace. All login into the same sulu installation.

**Scenario B:**

Each website is just a translation. So they share content and the structure of the tree but can use so translation and translation copy feature of sulu to provide e.g. `en`, `en_us`, `de`, ... version of website through different webspaces:

https://docs.sulu.io/en/2.4/book/localization.html

You configure the domain in the `<url>` for the specific localization. Through the language switcher you can change the content of the other language

**Scenario C:**

No content is shared between domainB.com, domainC.com, ... . So every domain has its own contacts, media, snippets. In this sceneria you the simpliest way is just have multiple installation (`/var/server/a`, `/server/website/b`, ...) of the same repository on your server but every installation have its own DATABASE_URL configured. If you need other webspace settings per deployment see [Overwrite webspaces config file path documentation here](https://docs.sulu.io/en/2.6/cookbook/override-dir-structure.html#overwrite-webspaces-config-file-path). 

----

So what you need to ask yourself what is the content of domainB.com and what is shared with domainC.com or what you do not want to share with domainC. Depending on that you need to go one of the 3 scenarios above.

**Reply by brachycera** [reply]
Wow, thanks for the fast and informative answer!!! 

Liebe Grüße von good-ol' Wien 🎡 nach Xi-🏔

**Comment by ibasaw** [comment]
i need scenario C

i have multiple domains (for now 2), i had installed SyliusThemeBundle, i got a themes directory with 2 different Themes.

All working fine when i stay on the same domain/url, but i need to have multiple domain/urls :

domain A => theme A
domain B => theme B


the goal is to have one theme different for each domain when i visiting site A or B.

i had created 2 xml inside config/webspaces, each file use different theme and key

how to do that ? what i miss ?

++

**Reply by ibasaw** [reply]
i succeed to make it work, by modifying in the webspace.xml the url for environnement  and language for each of my web space.

now i try to implement a solution to load a configuration from database, don't want to load the xml, how can i achive this and bypass the xml loading ?

domain A => theme A => load from db where HTTP_HOST = my key url A
domain B => theme B => load from db where HTTP_HOST = my key url B

**Reply by alexander-schranz** [reply]
You can change from where webspaces are loaded via a ENV variable which you can configure in your `.env.local` or as a real ENV var: https://docs.sulu.io/en/2.6/cookbook/override-dir-structure.html#overwrite-webspaces-config-file-path

**Reply by ibasaw** [reply]
i had succeed to do it with multi tenant solution by my own.

**Reply by alexander-schranz** [reply]
If you want to share your config let us know!
